src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Condition.cs

```diff
@@ -111,6 +114,7 @@ public unsafe bool Wait(int millisecondsTimeout)
             bool success = false;
             try
             {
+                Interlocked.Increment(ref _waitCount);
```

Comment:
I want the count to be increment only when the thread yields. Not sure if that's the case here.